### PR TITLE
MOSIP-23838 Added claim as Request Param in login.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

**Review by loganathan-sekaran — APPROVED** [review]

**Review by loganathan-sekaran — APPROVED** [review]

File: kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java [hunk 1]
```diff
@@ -1,49 +1,41 @@
...
-
 import com.fasterxml.jackson.databind.ObjectMapper;
-
 import io.mosip.kernel.authcodeflowproxy.api.constants.Constants;
 import io.mosip.kernel.authcodeflowproxy.api.constants.Errors;
-import io.mosip.kernel.authcodeflowproxy.api.dto.AccessTokenResponse;
-import io.mosip.kernel.authcodeflowproxy.api.dto.AccessTokenResponseDTO;
-import io.mosip.kernel.authcodeflowproxy.api.dto.IAMErrorResponseDto;
-import io.mosip.kernel.authcodeflowproxy.api.dto.MosipUserDto;
+import io.mosip.kernel.authcodeflowproxy.api.dto.*;
```
Comment (loganathan-sekaran) [inline kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java]: organize imports

File: kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java [hunk 2]
```diff
@@ -214,6 +215,44 @@ public AccessTokenResponseDTO loginRedirect(String state, String sessionState, S
...
+
+	private String getDataToSign() {
+		Map dataToSignMap = new LinkedHashMap();
+		String mosipResidentIDPClient = this.environment.getProperty(Constants.MOSIP_RESIDENT_IDP_CLIENT_ID);
+		dataToSignMap.put(Constants.SUB, mosipResidentIDPClient);
+		dataToSignMap.put(Constants.ISS, mosipResidentIDPClient);
+		dataToSignMap.put(Constants.AUD, Constants.BASE_URL);
+		dataToSignMap.put(Constants.EXP, getExpiryTime());
+		dataToSignMap.put(Constants.IAT, DateUtils.getUTCCurrentDateTimeString());
+		return CryptoUtil.encodeToPlainBase64(dataToSignMap.toString().getBytes());
```
Comment (loganathan-sekaran) [inline kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java]: Convert the map to JSON using ObjectMapper.

File: kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java [hunk 3]
```diff
@@ -214,6 +215,44 @@ public AccessTokenResponseDTO loginRedirect(String state, String sessionState, S
...
+		}
+	}
+
+	private String getDataToSign() {
+		Map dataToSignMap = new LinkedHashMap();
+		String mosipResidentIDPClient = this.environment.getProperty(Constants.MOSIP_RESIDENT_IDP_CLIENT_ID);
+		dataToSignMap.put(Constants.SUB, mosipResidentIDPClient);
+		dataToSignMap.put(Constants.ISS, mosipResidentIDPClient);
+		dataToSignMap.put(Constants.AUD, Constants.BASE_URL);
+		dataToSignMap.put(Constants.EXP, getExpiryTime());
```
Comment (loganathan-sekaran) [inline kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java:253]: Get the epochSeconds of the timestamp and then put into the map.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java (hunk 1 above)
Comment (kameshsr) [reply kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java]: Done.

File: kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java (hunk 2 above)
Comment (kameshsr) [reply kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java]: done.

File: kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java (hunk 3 above)
Comment (kameshsr) [reply kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java:253]: done.

**Review by loganathan-sekaran — APPROVED** [review]

**Review by vishwa-vyom — APPROVED** [review]
Approving based on the review done by Loga. Approval done for sake of merging.

**Pull request merged**